### PR TITLE
First architecture for plugins, with a dummy auth example

### PR DESCRIPTION
**Pull request by jvermillard**

### CODE REVIEW

File: src/github.com/airvantage/moxy/plugin/auth/auth.go [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+package auth
+
+import (
+	"encoding/gob"
+	"github.com/airvantage/moxy"
+	"log"
+	"os/exec"
+)
+
+// A plugin for authnetication, implements the Authneticator interface
```
Comment (phtrivier) [inline src/github.com/airvantage/moxy/plugin/auth/auth.go]: Typo. But at least, consistent (authnetication / Authneticator)

Comment (sebz) [inline src/github.com/airvantage/moxy/plugin/auth/auth.go]: @phtrivier the typo master fixes a typo :joy:

File: src/github.com/airvantage/moxy/plugin/auth/auth.go [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+		log.Fatal("Error running the auth plugin", err)
+		return res
+	}
+
+	res.dec = gob.NewDecoder(out)
+	res.enc = gob.NewEncoder(in)
+	return res
+}
+
+// implements the authenticator interfac by communication with the plugin command
```
Comment (phtrivier) [inline src/github.com/airvantage/moxy/plugin/auth/auth.go]: typo (interface)

**Comment by phtrivier** [comment]
As discussed, better use sockets than sys out/in.

File: src/github.com/airvantage/moxy/plugin/auth/auth.go (hunk 1 above)
Comment (jvermillard) [inline src/github.com/airvantage/moxy/plugin/auth/auth.go]: :hand:  the :trollface:

File: src/github.com/airvantage/moxy/plugin/auth/auth.go [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+// implements the Authenticator interface by communication with the plugin command
+func (ap *AuthPlugin) AuthUser(connection, user, password string) (moxy.AuthResult, error) {
+	var call AuthCall
+
+	call.Password = password
+	call.UserName = user
+
+	c, err := net.Dial("unix", "/tmp/auth.sock")
+	if err != nil {
+		panic(err)
```
Comment (phtrivier) [inline src/github.com/airvantage/moxy/plugin/auth/auth.go:45]: Will this happen whenever two auths are done concurrently (because of the lock ?)

Comment (jvermillard) [inline src/github.com/airvantage/moxy/plugin/auth/auth.go:45]: yes, it's like connecting to a TCP server (but it's a unix socket server)

**Pull request merged**